### PR TITLE
Added envoy as SNI dynamic forward proxy to cloud-config-container

### PR DESCRIPTION
**Pull request by apichick**
Added envoy as SNI dynamic forward proxy to cloud-config-container

---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

File: modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+This module depends on the cos-generic-metadata module being in the parent folder. If you change its location be sure to adjust the source attribute in main.tf.
+
+Logging and monitoring are enabled via the [Google Cloud Logging agent](https://cloud.google.com/container-optimized-os/docs/how-to/logging) configured for the instance via the `google-logging-enabled` metadata property, and the [Node Problem Detector](https://cloud.google.com/container-optimized-os/docs/how-to/monitoring) service started by default on boot.
+
+The module renders the generated cloud config in the `cloud_config` output, to be used in instances or instance templates via the `user-data` metadata.
+
+For convenience during development or for simple use cases, the module can optionally manage a single instance via the `test_instance` variable. If the instance is not needed the `instance*tf` files can be safely removed. Refer to the [top-level README](../README.md) for more details on the included instance.
```
Comment (wiktorn) [inline modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md:11]: This doesn't match PR content. Remove?

File: modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Containerized Envoy as SNI dynamic forward proxy on Container Optimized OS
+
+This module manages a `cloud-config` configuration that starts a containerized [Envoy SNI Dynamic forward proxy]https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/sni_dynamic_forward_proxy_filter) service on Container Optimized OS running on port 443.
```
Comment (wiktorn) [inline modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md:3]: I think, that maybe adding two sentences describing how this works, when it is useful, and involved risks, might be very helpful here.

**Review by wiktorn — APPROVED** [review]

File: modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md (hunk 1 above)
Comment (apichick) [reply modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md]: removed

File: modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md (hunk 2 above)
Comment (apichick) [reply modules/cloud-config-container/envoy-sni-dyn-fwd-proxy/README.md:3]: as discussed, we will add them once the apigee factory is there.

**Pull request merged**